### PR TITLE
Add registry and SkyDB to docs for nodejs

### PR DESCRIPTION
**Pull request by mrcnski**
# PULL REQUEST

## Overview

User noticed these docs were missing, although the methods are available now with `v2.6.0`.

### CODE REVIEW

File: source/includes/_registry.md [hunk 1]
```diff
@@ -68,6 +81,17 @@ Field | Description | Default
 }
 ```
 
+```javascript--node
+{
+  entry: {
+    datakey: "foo",
+    data: "bar",
+    revision: 0
```
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
    revision: 0n
```

File: source/includes/_registry.md [hunk 2]
```diff
@@ -68,6 +81,17 @@ Field | Description | Default
 }
 ```
 
+```javascript--node
+{
+  entry: {
+    datakey: "foo",
+    data: "bar",
```
Comment (interstella-5555) [inline source/includes/_registry.md]: isn't `data` an `Uint8Array` ?

File: source/includes/_registry.md (hunk 1 above)
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
    revision: 0n
```

File: source/includes/_registry.md [hunk 3]
```diff
@@ -94,6 +118,22 @@ async function setEntryExample() {
...
 
+```javascript--node
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { privateKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+const data = "bar";
+const revision = 0;
```
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
const revision = BigInt(0);
```

File: source/includes/_registry.md [hunk 4]
```diff
@@ -126,3 +166,53 @@ Field | Type | Description
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+
+(async () => {
+  const url = await client.registry.getEntryUrl(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
(async () => {
  const url = await client.registry.getEntryUrl(publicKey, dataKey);
})();
```

File: source/includes/_registry.md [hunk 5]
```diff
@@ -126,3 +166,53 @@ Field | Type | Description
...
+
+const dataKey = "foo";
+
+function getEntryExample() {
+  try {
+    const url = client.registry.getEntryUrl(publicKey, dataKey);
+  } catch (error) {
+    console.log(error);
+  }
+}
```
Comment (interstella-5555) [inline source/includes/_registry.md]: Any reason to wrap this in a function with throw-catch ? I think the example should be as simple as possible like

```suggestion
const url = client.registry.getEntryUrl(publicKey, dataKey);
```

File: source/includes/_registry.md [hunk 6]
```diff
@@ -94,6 +118,22 @@ async function setEntryExample() {
...
+const { privateKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+const data = "bar";
+const revision = 0;
+const entry = { dataKey, data, revision };
+
+(async () => {
+  await client.registry.setEntry(privateKey, entry);
+}
```
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
(async () => {
  await client.registry.setEntry(privateKey, entry);
})();
```

File: source/includes/_registry.md [hunk 7]
```diff
@@ -36,6 +36,19 @@ async function getEntryExample() {
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+
+(async () => {
+  const { entry, signature } = await client.registry.getEntry(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/includes/_registry.md]:
```suggestion
(async () => {
  const { entry, signature } = await client.registry.getEntry(publicKey, dataKey);
})();
```

File: source/includes/_skydb.md [hunk 8]
```diff
@@ -32,6 +32,19 @@ async function getJSONExample() {
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "myApp";
+
+(async () => {
+  const { data, dataLink } = await client.db.getJSON(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/includes/_skydb.md]:
```suggestion
(async () => {
  const { data, dataLink } = await client.db.getJSON(publicKey, dataKey);
})();
```

File: source/includes/_skydb.md [hunk 9]
```diff
@@ -78,51 +100,32 @@ async function setJSONExample() {
...
 
-function getEntryExample() {
-  try {
-    const url = client.registry.getEntryUrl(publicKey, dataKey);
-  } catch (error) {
-    console.log(error);
-  }
+(async () => {
+  await client.db.setJSON(privateKey, dataKey, json);
 }
```
Comment (interstella-5555) [inline source/includes/_skydb.md]:
```suggestion
(async () => {
  await client.db.setJSON(privateKey, dataKey, json);
})();
```

File: source/v4/includes/_registry.md [hunk 10]
```diff
@@ -36,6 +36,19 @@ async function getEntryExample() {
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+
+(async () => {
+  const { entry, signature } = await client.registry.getEntry(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]:
```suggestion
(async () => {
  const { entry, signature } = await client.registry.getEntry(publicKey, dataKey);
})();
```

File: source/v4/includes/_registry.md [hunk 11]
```diff
@@ -68,6 +81,17 @@ Field | Description | Default
 }
 ```
 
+```javascript--node
+{
+  entry: {
+    datakey: "foo",
+    data: "bar",
+    revision: 0
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]:
```suggestion
    revision: 0n
```

File: source/v4/includes/_registry.md [hunk 12]
```diff
@@ -68,6 +81,17 @@ Field | Description | Default
 }
 ```
 
+```javascript--node
+{
+  entry: {
+    datakey: "foo",
+    data: "bar",
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]: AFAIK this is a Uint8Array

File: source/v4/includes/_registry.md [hunk 13]
```diff
@@ -94,6 +118,22 @@ async function setEntryExample() {
...
+const { privateKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+const data = "bar";
+const revision = 0;
+const entry = { dataKey, data, revision };
+
+(async () => {
+  await client.registry.setEntry(privateKey, entry);
+}
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]:
```suggestion
(async () => {
  await client.registry.setEntry(privateKey, entry);
})();
```

File: source/v4/includes/_registry.md [hunk 14]
```diff
@@ -94,6 +118,22 @@ async function setEntryExample() {
...
 
+```javascript--node
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { privateKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+const data = "bar";
+const revision = 0;
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]:
```suggestion
const revision = BigInt(0);
```

File: source/v4/includes/_registry.md [hunk 15]
```diff
@@ -94,6 +118,22 @@ async function setEntryExample() {
...
 ```
 
+```javascript--node
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { privateKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+const data = "bar";
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]: it can be a string or only Uint8Array ?

File: source/v4/includes/_registry.md [hunk 16]
```diff
@@ -126,3 +166,53 @@ Field | Type | Description
...
+
+const dataKey = "foo";
+
+function getEntryExample() {
+  try {
+    const url = client.registry.getEntryUrl(publicKey, dataKey);
+  } catch (error) {
+    console.log(error);
+  }
+}
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]: Any reason to wrap this in a function with throw-catch ? I think the example should be as simple as possible like

```suggestion
const url = client.registry.getEntryUrl(publicKey, dataKey);
```

File: source/v4/includes/_registry.md [hunk 17]
```diff
@@ -126,3 +166,53 @@ Field | Type | Description
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "foo";
+
+(async () => {
+  const url = await client.registry.getEntryUrl(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/v4/includes/_registry.md]:
```suggestion
(async () => {
  const url = await client.registry.getEntryUrl(publicKey, dataKey);
})();
```

File: source/v4/includes/_skydb.md [hunk 18]
```diff
@@ -32,6 +32,19 @@ async function getJSONExample() {
...
+const { SkynetClient, genKeyPairFromSeed } = require("@skynetlabs/skynet-nodejs");
+
+const client = new SkynetClient();
+const { publicKey } = genKeyPairFromSeed("this seed should be fairly long for security");
+
+const dataKey = "myApp";
+
+(async () => {
+  const { data, dataLink } = await client.db.getJSON(publicKey, dataKey);
+}
```
Comment (interstella-5555) [inline source/v4/includes/_skydb.md]:
```suggestion
(async () => {
  const { data, dataLink } = await client.db.getJSON(publicKey, dataKey);
})();
```

File: source/v4/includes/_skydb.md [hunk 19]
```diff
@@ -78,51 +100,32 @@ async function setJSONExample() {
...
 
-function getEntryExample() {
-  try {
-    const url = client.registry.getEntryUrl(publicKey, dataKey);
-  } catch (error) {
-    console.log(error);
-  }
+(async () => {
+  await client.db.setJSON(privateKey, dataKey, json);
 }
```
Comment (interstella-5555) [inline source/v4/includes/_skydb.md]:
```suggestion
(async () => {
  await client.db.setJSON(privateKey, dataKey, json);
})();
```

**Review by interstella-5555 — CHANGES_REQUESTED** [review]

**Review by interstella-5555 — DISMISSED** [review]

**Review by ro-tex — APPROVED** [review]

**Review by interstella-5555 — APPROVED** [review]

**Review by dghelm — APPROVED** [review]
Looks good! Much needed refresh on latest features.

Follow up note: we should probably add tryFiles details to uploading a directory and link to docs for more information: https://docs.skynetlabs.com/skynet-topics/framework-flexible-routing

**Comment by mrcnski** [comment]
@dghelm Thanks, wrote up an issue!

https://linear.app/skynetlabs/issue/SKY-1238/add-tryfiles-details-to-docs

**Pull request merged**